src/puppetlabs/puppetdb/command.clj

```diff
@@ -215,8 +224,22 @@
       (scf-storage/replace-facts! fact-data))
     (log/infof "[%s] [%s] %s" id (command-names :replace-facts) certname)))
 
-;; Node deactivation
+(defmethod process-command! [(command-names :replace-facts) 3]
+  [command config]
+  (-> command
+      (assoc :version 4)
+      (update :payload #(set/rename-keys % {:producer-timestamp :producer_timestamp
+                                            :name :certname}))
+      (process-command! config)))
+
+(defmethod process-command! [(command-names :replace-facts) 2]
+  [{:keys [annotations] :as command} config]
+  (-> command
+      (assoc :version 3)
+      (assoc-in [:payload :producer-timestamp] (:received annotations))
+      (process-command! config)))
 
+;; Node deactivation
```

Comment:
Forgot that we had bumped that, I'll figure out what we need to support here and add it